gateway-ha/src/test/java/com/lyft/data/gateway/ha/router/TestRoutingGroupSelector.java

```diff
@@ -0,0 +1,24 @@
+package com.lyft.data.gateway.ha.router;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import javax.servlet.http.HttpServletRequest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+@Test
+public class TestRoutingGroupSelector {
+  public void testByRoutingGroupHeader() {
+    HttpServletRequest mockRequest = mock(HttpServletRequest.class);
+
+    // If the header is present the routing group is the value of that header.
+    when(mockRequest.getHeader("X-Presto-Routing-Group")).thenReturn("batch_backend");
```

Comment:
You can use the `ROUTING_GROUP_HEADER` constant here too.

Reply:
Good call, done!